### DISCUSSION TITLE
Programing The78mole KM271 0.0.6

### DISCUSSION THREAD

**Discussion by SteanMal** [dewenni/ESP_Buderus_KM271#32] [General]
Hej,
I have a problem with programming KM271 board. I was trying to flash ESP32 with file "esp_buderus_km271_v3_2_0.bin".
Software is not working (setup mode), but works if I program it with The78mole - ESPhome software.
Should this board work with bin file or should I program it in different way?

SM

**Comment by dewenni** [comment]
Hi @SteanMal 
what do you mean with :
> Software is not working (setup mode)

Does the software starts the local access-point with SSID `ESP-Buderus-KM271`?
Or does this not happen?

What about the LED status of the board?
The green LED should blink if it is in setup-mode

**Reply by dewenni** [reply]
maybe you have to "power off" => "power on" the board once after flashing.

**Reply by SteanMal** [reply]
Hej @dewenni 
Software don't start local access-point.
None of LED is on, and none of them are blinking. I program ESP32 with esptool in Linux.
After flashing and restarting the power this error appear, 
␊rst:0x10 (RTCWDT_RTC_RESET),boot:0x17 (SPI_FAST_FLASH_BOOT)␍
␊flash read err, 1000␍
␊ets_main.c 371 ␍
␊ets Jun  8 2016 00:22:57␍
(serial debug)

SM

**Comment by the78mole** [comment]
Hi Friends,

there is only one possibly relevant change in hardware. One LED was relocated from IO23 to IO17 (see image). 

![image](https://github.com/dewenni/ESP_Buderus_KM271/assets/7723396/8533a8d1-e6c4-4ceb-9e73-b297952a987f)

But I do not believe, that this is related to the issue you encounter :-(
Could you give us a hint, what exact hardware you have (order options, date of order) and what you already have connected to the board. Then I could maybe give some better hints on where to look for issues. There are some strapping pins on ESP32 that serve as boot select pins...

Regards,
Daniel

**Reply by dewenni** [reply]
thanks for helping Daniel!
After some short research with the given error message, for me it seems that there is a problem with the flash?!
But to be honest, I am not an expert in debugging and interpretation of this error messages.

In general my software is working fine on the new board. I tested it again yesterday on purpose.
I don´t think that it makes a difference, but maybe you can also test the newest release from today => [v3.2.1](https://github.com/dewenni/ESP_Buderus_KM271/releases/tag/v3.2.1)

**Comment by the78mole** [comment]
@SteanMal 
could it be, that you used a wrong flashing tool? At least for ESPhome, you have the Option with the factory-image and the standard ESPhome image. The latter one can be used only with ESPhome flasher, the first one is suitable for used with any flasher starting at address 0x0. On the other hand, I don't beleive in a HW-Flash problem. Usually, flashers do a checksum check after flashing. This should already report a problem, when something get wrong on this side
If you provide a bit more information, I'm sure, I/we can help.
Regards,
Daniel

**Comment by SteanMal** [comment]
@dewenni @the78mole 
I program the board successfully. I compiled simple sketch in Arduino IDE with AsyncElegantOTA. Next I flash 
esp_buderus_km271_v3_2_0.bin using ElegantOTA web interface. Now board 0.0.6 is working fine.
Earlier i was trying to flash it using esptool from address 0x0 and it doesn't work.
Maybe bootloader is missing from 0x0 to 0x1000 address in esp_buderus_km271_v3_2_0.bin.
I don't test esp_buderus_km271_v3_2_1.bin version.
Regards,
SM

**Reply by dewenni** [reply]
thanks for your feedback and good to hear that it is working now.

I am not familiar with the esp flash tools at all. Normally I download the code directly with the IDE.
So I can not add any more information on this. The .bin File is the File that is automatically generated with every build in PlatformIO IDE.

But you should now also be able to use the newest release v3.2.1.
I have did some improvements regarding the automatic reconnect of WiFi and mqtt.
